code/game/mecha/equipment/weapons/weapons.dm

```diff
@@ -12,15 +13,30 @@
 			return 1
 	return 0
 
+/obj/item/mecha_parts/mecha_equipment/weapon/become_damaged()
+	if(!damaged)
+		damaged = 1
+		desc += "/nIt doesn't look to be in the best shape."
```

Comment:
I only wanted to add the damaged var for things that I actually made an effect for, should I just put it on obj/item or something?